src/components/ordersTable/utils/statusChangeHandlers.js

```diff
@@ -0,0 +1,23 @@
+import { doc, updateDoc } from 'firebase/firestore'
+import { db } from "./../../../api/firebase"
+
+export const changeEditStatus = (event, isEditStatusActive, setIsEditStatusActive) => {
+    setIsEditStatusActive(!isEditStatusActive)
+    event.stopPropagation()
+}
+
+export const cancelStatusChange = (event, isEditStatusActive, setIsEditStatusActive) => {
+    changeEditStatus(event, isEditStatusActive, setIsEditStatusActive)
+}
+
+export const confirmStatusChange = (event, order, orderStatusSelectValue, isEditStatusActive, setIsEditStatusActive) => {
+    changeEditStatus(event, isEditStatusActive, setIsEditStatusActive)
+    updateOrderStatus(order, orderStatusSelectValue)
+}
+
+const updateOrderStatus = (order, orderStatusSelectValue) => {
+    order.status = orderStatusSelectValue
```

Comment:
moze da się to przerobic zeby update tego statusu był wykonywany za pomocą callbacka (czyli przekazujemy do funkcji updateOrderStatus jakas inna funkcję jako argument) a ta funkcja zeby updateowala order tam gdzie on jest zdefiniowany? 

modyfikowanie obiektów poprzez referencje raczej nie jest dobrym patternem, często w reactcie jesli tak robimy mozemy gubic stan itd.